### DISCUSSION TITLE
How to prevent starting selection with specific condition on Firefox and Safari browsers?

### DISCUSSION THREAD

**Discussion by namlk-hamsa-dev** [simonwep/viselect#147] [Q&A] [answered]
I use @viselect/react. Since I want to prevent starting selection for cards with class 'matrix-item' I set the onBeforeStart event with the following function:

`  const handleBeforeStart = (e: any) => {
    return !e.event.path.some((item: any) => item.classList && item.classList.contains('matrix-item'));
  };`

This works fine on chrome. But when I run my code on other browsers (firefox/safari) this function doesn't work because the event.path array returned from onBeforeStart event doesn't exist.

I tried to search in the Issues list but I haven't found any ideas yet. So is there any way to solve this problem? Thanks

**Comment by simonwep** [comment, marked as answer]
Try using [composedPath](https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath), I'll update the example since .path is chromium only.

Also mentioned in #148, thanks @KKondratenya for the quick answer!
